examples/step-75/step-75.cc

```diff
@@ -0,0 +1,1575 @@
+    }
+
+    virtual double value(const Point<dim> & p,
+                         const unsigned int component = 0) const override;
+
+    virtual Tensor<1, dim>
+    gradient(const Point<dim> & p,
+             const unsigned int component = 0) const override;
+
+    virtual double laplacian(const Point<dim> & p,
+                             const unsigned int component = 0) const override;
+
+  private:
+    const double alpha;
+  };
+
+
+
+  template <int dim>
+  double Solution<dim>::value(const Point<dim> &p,
+                              const unsigned int /*component*/) const
+  {
+    const std::array<double, dim> p_sphere =
+      GeometricUtilities::Coordinates::to_spherical(p);
+
+    return std::pow(p_sphere[0], alpha) * std::sin(alpha * p_sphere[1]);
+  }
+
+
+
+  template <int dim>
+  Tensor<1, dim> Solution<dim>::gradient(const Point<dim> &p,
+                                         const unsigned int /*component*/) const
+  {
+    const std::array<double, dim> p_sphere =
+      GeometricUtilities::Coordinates::to_spherical(p);
+
+    std::array<double, dim> ret_sphere;
+    // only for polar coordinates
+    const double fac = alpha * std::pow(p_sphere[0], alpha - 1);
+    ret_sphere[0]    = fac * std::sin(alpha * p_sphere[1]);
+    ret_sphere[1]    = fac * std::cos(alpha * p_sphere[1]);
+
+    // transform back to cartesian coordinates
+    // by considering polar unit vectors
+    Tensor<1, dim> ret;
+    ret[0] = ret_sphere[0] * std::cos(p_sphere[1]) -
+             ret_sphere[1] * std::sin(p_sphere[1]);
+    ret[1] = ret_sphere[0] * std::sin(p_sphere[1]) +
+             ret_sphere[1] * std::cos(p_sphere[1]);
+    return ret;
+  }
+
+
+
+  template <int dim>
+  double Solution<dim>::laplacian(const Point<dim> & /*p*/,
+                                  const unsigned int /*component*/) const
+  {
+    return 0.;
+  }
+
+
+
+  // @sect3{The <code>LaplaceOperator</code> class template}
+
+  // Operator class template for solver.
+  template <int dim, typename number>
+  class LaplaceOperator
+  {
+  public:
+    using VectorType = LinearAlgebra::distributed::Vector<number>;
+
+    virtual void reinit(const hp::MappingCollection<dim> &mapping_collection,
+                        const DoFHandler<dim> &           dof_handler,
+                        const hp::QCollection<dim> &      quadrature_collection,
+                        const AffineConstraints<number> & constraints,
+                        VectorType &                      system_rhs) = 0;
+
+
+    virtual void vmult(VectorType &dst, const VectorType &src) const = 0;
+
+    virtual const TrilinosWrappers::SparseMatrix &get_system_matrix() const = 0;
+
+    virtual void initialize_dof_vector(VectorType &vec) const = 0;
+  };
+
+
+
+  template <int dim, typename number>
+  class LaplaceOperatorMatrixBased : public LaplaceOperator<dim, number>
+  {
+  public:
+    using VectorType = LinearAlgebra::distributed::Vector<number>;
```

Comment:
Might be cleaner to import the type from the base class -- just one place where one would have to write it out.